### PR TITLE
WIP: Monerodart integration to staging

### PR DESCRIPTION
**Pull request by sneurlax**
This is based on https://github.com/cypherstack/stack_wallet/pull/818: I rebased `MrCyjaneK:monerodart` on `cypherstack:staging`.  #818 can be merged now.

 - [x] Rebase MrCyjaneK's contributions from https://github.com/cypherstack/stack_wallet/pull/818 on latest `cypherstack:staging`
 - [ ] Port build instructions from https://github.com/cypherstack/stack_wallet/pull/818 into `docs/building.md` and/or script appropriately
 - [ ] Complete review

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	url = https://github.com/cypherstack/flutter_liblelantus.git
 [submodule "crypto_plugins/monero_c"]
 	path = crypto_plugins/monero_c
-	url = https://git.mrcyjanek.net/mrcyjanek/monero_c
+	url = git@github.com:cypherstack/monero_c
```
Comment (MrCyjaneK) [inline .gitmodules:9]: Hey! The upstream source code is also available at https://github.com/mrcyjanek/monero_c

Comment (MrCyjaneK) [inline .gitmodules:9]: Hey! The upstream source code is also available at https://github.com/mrcyjanek/monero_c

**Comment by sneurlax** [comment]
We can revert to your github repo and pin it to an audited commit hash.

I do need to change those SSH-based git@github.com URLs to typical HTTPS links, though.

File: scripts/windows/build_all.sh [hunk 2]
```diff
@@ -18,8 +18,10 @@ sudo apt install -y gcc g++ gperf # Can be removed (deps are listed in docs/buil
 pushd ../../crypto_plugins/monero_c
     ./apply_patches.sh monero
     ./apply_patches.sh wownero
-    ./build_single.sh monero x86_64-w64-mingw32 -j$(nproc)
-    ./build_single.sh wownero x86_64-w64-mingw32 -j$(nproc)
+    sudo ./build_single.sh monero x86_64-w64-mingw32 -j$(nproc)
+    # WSL2 needs superuser permissions related to error:
+    # update-alternatives: error creating symbolic link ...: Permission denied
+    sudo ./build_single.sh wownero x86_64-w64-mingw32 -j$(nproc)
```
Comment (MrCyjaneK) [inline scripts/windows/build_all.sh:24]: SUDO=sudo ./build_single.sh wownero x86_64-w64-mingw32 -j$(nproc)

https://git.mrcyjanek.net/mrcyjanek/monero_c/src/commit/a032a20221579d5fb9445b370eec1b6bd4ff653b/build_single.sh#L87 (I need to change upstream for i686, but it is not used there)

File: docs/building.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ Make a Pixel 4 (API 30) x86_64 emulator with 2GB of storage space for emulation
 
 Install basic dependencies
 ```
-sudo apt-get install libssl-dev curl unzip automake build-essential file pkg-config git python libtool libtinfo5 cmake libgit2-dev clang libncurses5-dev libncursesw5-dev zlib1g-dev llvm python3-distutils
+sudo apt-get install libssl-dev curl unzip automake build-essential file pkg-config git python libtool libtinfo5 cmake libgit2-dev clang libncurses5-dev libncursesw5-dev zlib1g-dev llvm python3-distutils gcc g++ gperf
```
Comment (MrCyjaneK) [inline docs/building.md:34]: libtinfo5 for android

File: scripts/windows/build_all.sh (hunk 2 above)
Comment (sneurlax) [reply scripts/windows/build_all.sh:24]: Sorry, I'm not sure how to integrate this comment?

**Comment by sneurlax** [comment]
Recreate using https://github.com/cypherstack/stack_wallet/pull/831

**Pull request closed without merging**